RecoEgamma/ElectronIdentification/src/ElectronMVAEstimatorRun2Fall17.cc

```diff
@@ -0,0 +1,428 @@
+#include "RecoEgamma/ElectronIdentification/interface/ElectronMVAEstimatorRun2Fall17.h"
+
+ElectronMVAEstimatorRun2Fall17::ElectronMVAEstimatorRun2Fall17(const edm::ParameterSet& conf, bool withIso):
+  AnyMVAEstimatorRun2Base(conf),
+  tag_(conf.getParameter<std::string>("mvaTag")),
+  MethodName_("BDTG method"),
+  beamSpotLabel_(conf.getParameter<edm::InputTag>("beamSpot")),
+  conversionsLabelAOD_(conf.getParameter<edm::InputTag>("conversionsAOD")),
+  conversionsLabelMiniAOD_(conf.getParameter<edm::InputTag>("conversionsMiniAOD")),
+  rhoLabel_(edm::InputTag("fixedGridRhoFastjetAll")) {
+
+  const std::vector <std::string> weightFileNames
+    = conf.getParameter<std::vector<std::string> >("weightFileNames");
+
+  if( (int)(weightFileNames.size()) != nCategories )
+    throw cms::Exception("MVA config failure: ")
+      << "wrong number of weightfiles" << std::endl;
+
+  gbrForest_s.clear();
+  // Create a TMVA reader object for each category
+  for(int i=0; i<nCategories; i++){
+
+    // Use unique_ptr so that all readers are properly cleaned up
+    // when the vector clear() is called in the destructor
+
+    edm::FileInPath weightFile( weightFileNames[i] );
+    gbrForest_s.push_back( GBRForestTools::createGBRForest( weightFile ) );
+  }
+
+  this->withIso = withIso;
```

Comment:
naming private member variables with a trailing underscore would fix this issue :)